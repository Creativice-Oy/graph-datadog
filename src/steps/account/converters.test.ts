import { getMockAccount } from '../../../test/mocks';
import {
  createAccountEntity,
  createOrganizationAccountEntity,
} from './converters';

describe('#createAccountEntity', () => {
  test('should convert to entity', () => {
    expect(createAccountEntity('Local Integration')).toMatchSnapshot();
  });
});

describe('#createOrganizationAccountEntity', () => {
  test('should convert to entity', () => {
    expect(createOrganizationAccountEntity(getMockAccount())).toMatchSnapshot();
  });
});
