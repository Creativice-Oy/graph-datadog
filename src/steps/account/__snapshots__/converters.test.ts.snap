// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#createAccountEntity should convert to entity 1`] = `
Object {
  "_class": Array [
    "Account",
  ],
  "_key": "datadog_account:local-integration",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "name": "Local Integration",
      },
    },
  ],
  "_type": "datadog_account",
  "createdOn": undefined,
  "displayName": "Local Integration",
  "name": "Local Integration",
}
`;

exports[`#createOrganizationAccountEntity should convert to entity 1`] = `
Object {
  "_class": Array [
    "Account",
  ],
  "_key": "datadog_account:96bd2d8e-4625-11ec-a18c-da7ad0900002",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "billing": Object {
          "type": "cc",
        },
        "name": "JupiterOne Inc.",
        "publicId": "96bd2d8e-4625-11ec-a18c-da7ad0900002",
        "settings": Object {
          "privateWidgetShare": false,
          "saml": Object {
            "enabled": false,
          },
          "samlAutocreateAccessRole": "st",
          "samlAutocreateUsersDomains": Object {
            "domains": Array [],
            "enabled": false,
          },
          "samlCanBeEnabled": false,
          "samlIdpInitiatedLogin": Object {
            "enabled": false,
          },
          "samlIdpMetadataUploaded": false,
          "samlStrictMode": Object {
            "enabled": false,
          },
        },
        "subscription": Object {
          "type": "pro",
        },
      },
    },
  ],
  "_type": "datadog_account",
  "billingType": "cc",
  "createdOn": undefined,
  "description": "",
  "displayName": "JupiterOne Inc.",
  "name": "JupiterOne Inc.",
  "settings.privateWidgetShare": false,
  "settings.samlAutocreateAccessRole": "st",
  "settings.samlAutocreateUsersDomainsEnabled": false,
  "settings.samlCanBeEnabled": false,
  "settings.samlEnabled": false,
  "settings.samlIdpEndpoint": undefined,
  "settings.samlIdpInitiatedLoginEnabled": false,
  "settings.samlIdpMetadataUploaded": false,
  "settings.samlLoginUrl": undefined,
  "settings.samlStrictModeEnabled": false,
  "subscriptionType": "pro",
}
`;
